scripts/twister

```diff
@@ -195,225 +190,15 @@ if not ZEPHYR_BASE:
-        # Show only the discards that apply to the selected platforms on the
-        # command line
-
-        for i in tplan.instances.values():
-            if i.status == "filtered":
-                if options.platform and i.platform.name not in options.platform:
-                    continue
-                logger.debug(
-                    "{:<25} {:<50} {}SKIPPED{}: {}".format(
-                        i.platform.name,
-                        i.testsuite.name,
-                        Fore.YELLOW,
-                        Fore.RESET,
-                        i.reason))
-
-    if options.report_excluded:
-        tplan.report_excluded_tests()
-        return 0
-
-    report = Reporting(tplan, env)
-    plan_file = os.path.join(options.outdir, "testplan.json")
-    if not os.path.exists(plan_file):
-        report.json_report(plan_file)
-
-    if options.save_tests:
-        report.json_report(options.save_tests)
-        return 0
-
-    if options.device_testing and not options.build_only:
-        print("\nDevice testing on:")
-        hwm.dump(filtered=tplan.selected_platforms)
-        print("")
-
-    if options.dry_run:
-        duration = time.time() - start_time
-        logger.info("Completed in %d seconds" % (duration))
-        return 0
-
-    if options.short_build_path:
-        tplan.create_build_dir_links()
-
-    runner = TwisterRunner(tplan.instances, tplan.testsuites, env)
-    runner.duts = hwm.duts
-    runner.run()
-
-    # figure out which report to use for size comparison
-    report_to_use = None
-    if options.compare_report:
-        report_to_use = options.compare_report
-    elif options.last_metrics:
-        report_to_use = previous_results_file
-
-    report.footprint_reports(report_to_use,
-                            options.show_footprint,
-                            options.all_deltas,
-                            options.footprint_threshold,
-                            options.last_metrics)
-
-    duration = time.time() - start_time
-
-    runner.results.summary()
-
-    report.summary(runner.results, options.disable_unrecognized_section_test, duration)
-
-    if options.coverage:
-        if not options.build_only:
-            run_coverage(tplan, options)
-        else:
-            logger.info("Skipping coverage report generation due to --build-only.")
-
-    if options.device_testing and not options.build_only:
-        hwm.summary(tplan.selected_platforms)
-
-    report.save_reports(options.report_name,
-                       options.report_suffix,
-                       options.report_dir,
-                       options.no_update,
-                       options.platform_reports
-                       )
-
-    if options.package_artifacts:
-        artifacts = Artifacts(env)
-        artifacts.package()
-
-    logger.info("Run completed")
-    if runner.results.failed or runner.results.error or (tplan.warnings and options.warnings_as_errors):
-        return 1
-
-    return 0
-
```

Comment:
why removing this code?

Reply:
@hakehuang - didn't remove, I created a function for this code which moved here
scripts/pylib/twister/twisterlib/twister_main.py

I broke this into smaller functions to be able to run twister from west.